packages/flutter/lib/src/gestures/tap.dart

```diff
@@ -174,6 +174,9 @@ class TapGestureRecognizer extends PrimaryPointerGestureRecognizer {
       _finalPosition = event.position;
       _checkUp();
     } else if (event is PointerCancelEvent) {
+      if (_sentTapDown && onTapCancel != null) {
```

Comment:
FWIW, I think there might be some value in being consistent with 190 (which I think means it's possible for a single tap recognizer to get nothing but just a single cancel callback if it won by a sweep and moved past the slop without checking for _sentTapDown).